src/index.ts

```diff
@@ -38,17 +83,38 @@ function mergeToV3ParamsWithDefaults(params?: Partial<V3Params>): V3Params {
   if (!params) {
     return v3Defaults
   }
+
+  if (typeof params.salt === 'string') {
+    params.salt = Buffer.from(validateHexString('salt', params.salt), 'hex')
```

Comment:
I just noticed that in `ethereumjs-util` we pad the string to even length, so that if implicitly e.g. a `0x01` byte was converted to `0x1`, it would be covered:

https://github.com/ethereumjs/ethereumjs-util/blob/3b1085059194b02354177d334f89cd82a5187883/src/bytes.ts#L76

Not sure whether that applies here.

Reply:
It's a possibility, for sure, though I wonder if it would end up masking legit user errors more than providing shorthand/convenience.